src/routing/history/StateHistory.ts

```diff
@@ -53,6 +53,9 @@ export class StateHistory implements HistoryInterface {
 	}
 
 	public set(path: string) {
+		if (path === this._window.location.pathname) {
```

Comment:
@xiaohulu Do you think we should move the logic from `_onChange` that does a check against the current and previous value to `set` function? 

```ts
const pathName = this._window.location.pathname.replace(/\/$/, '');
const value = stripBase(this._base, pathName + this._window.location.search);
if (this._current === value) {
	return;
}
```